tango_simlib/utilities/fandango_json_parser.py

```diff
@@ -176,9 +181,9 @@ def _extract_attribute_events_properties(self, attribute_events_configuration):
         for event_props in attribute_events_configuration.values():
             events_properties.update(event_props)
 
-        # Rename key 'period' to 'event_period'.
+        # Assign the value of 'period' to 'event_period' as it actually
+        # represents 'event_period' of the attribute.
         events_properties["event_period"] = events_properties["period"]
-        events_properties.pop("period")
```

Comment:
We thought about that and realised that we might need to make that change across all file parsers because all of them were using `period` to represent `polling_period`. But reading through the https://pytango.readthedocs.io/en/stable/server_api/attribute.html?highlight=set_default_pro#tango.UserDefaultAttrProp.set_event_period it seems `event_period` is represented by `period` here. It's a bit confusing.
So we will make that change to replace `period` with `polling_period`.

Reply:
> But reading through the https://pytango.readthedocs.io/en/stable/server_api/attribute.html?highlight=set_default_pro#tango.UserDefaultAttrProp.set_event_period it seems `event_period` is represented by `period` here. It's a bit confusing.

That link doesn't say much to me.  There is a method to modify the period, and it takes a parameter called `def_period`.  Where do you see that `period` is being used?

Reply:
Well looking at [`set_period`](https://pytango.readthedocs.io/en/stable/server_api/attribute.html?highlight=set_default_pro#tango.UserDefaultAttrProp.set_period) they say we should use [`set_event_period`](https://pytango.readthedocs.io/en/stable/server_api/attribute.html?highlight=set_default_pro#tango.UserDefaultAttrProp.set_event_period). And when I look at the the `UserDefaultAttrProp`attributes it doesn't have `event_period` but `period`:

```python
ipdb> attr_props
<tango._tango.UserDefaultAttrProp object at 0x7f9716b402f0>
ipdb> dir(attr_props)
[.., '_set_enum_labels', 'abs_change', 'archive_abs_change', 'archive_period', 'archive_rel_change', 'delta_t', 'delta_val', 'description', 'display_unit', 'enum_labels', 'format', 'label', 'max_alarm', 'max_value', 'max_warning', 'min_alarm', 'min_value', 'min_warning', 'period', 'rel_change', 'set_abs_change', 'set_archive_abs_change', 'set_archive_event_abs_change', 'set_archive_event_period', 'set_archive_event_rel_change', 'set_archive_period', 'set_archive_rel_change', 'set_delta_t', 'set_delta_val', 'set_description', 'set_display_unit', 'set_enum_labels', 'set_event_abs_change', 'set_event_period', 'set_event_rel_change', 'set_format', 'set_label', 'set_max_alarm', 'set_max_value', 'set_max_warning', 'set_min_alarm', 'set_min_value', 'set_min_warning', 'set_period', 'set_rel_change', 'set_standard_unit', 'set_unit', 'standard_unit', 'unit']
```

Reply:
OK.  As long as it works 🙂